framework/managers/MqttConnectionManager.cpp

```diff
@@ -61,8 +61,12 @@ void MqttConnectionManager::onDisconnected(TcpClient &client, bool flag) {
         LOG.log("Unreachable");
     }
 
-    this->state.set(State::DISCONNECTED);
-    this->reconnectTimer.start();
+    if (this->getState() != State::CONNECTING) {
+        this->state.set(State::DISCONNECTED);
+        this->reconnectTimer.start();
+    } else {
+        LOG.log("Ignored disconnect while connecting");
```

Comment:
As this is a workaround we should use a log message which is more alerting. At least prefix with `Error: ` or something like that.